Install networkx when testing functorch

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #81404
* __->__ #81403

networkx is a required testing dependency. We're working on getting rid
of the dependency though (because ultimately we don't want PyTorch to
take a dependency on it).

Test Plan:
- wait for the functorch test shard in pytorch CI

zou3519 committed:
Install networkx when testing functorch

networkx is a required testing dependency. We're working on getting rid
of the dependency though (because ultimately we don't want PyTorch to
take a dependency on it).

Test Plan:
- wait for the functorch test shard in pytorch CI

[ghstack-poisoned]

zou3519 committed:
Update on "Install networkx when testing functorch"


networkx is a required testing dependency. We're working on getting rid
of the dependency though (because ultimately we don't want PyTorch to
take a dependency on it).

Test Plan:
- wait for the functorch test shard in pytorch CI

[ghstack-poisoned]

malfet commented on .jenkins/pytorch/test.sh:
```diff
@@ -586,6 +586,8 @@ test_dynamo() {
 }
 
 test_functorch() {
+  # functorch testing dependency
+  pip install networkx
```
To avoid flaky network failure, can you please instead add it to https://github.com/pytorch/pytorch/blob/master/.circleci/docker/requirements-ci.txt

This way it will be baked into docker image

zou3519 replied:
Yes, will do, thanks for the suggestion

malfet replied:
You can delete it from this file if you've added it to requirements-ci.txt, don't you?

zou3519 replied:
Good point, let me submit a follow-up

zou3519 committed:
Update on "Install networkx when testing functorch"


networkx is a required testing dependency. We're working on getting rid
of the dependency though (because ultimately we don't want PyTorch to
take a dependency on it).

Test Plan:
- wait for the functorch test shard in pytorch CI

[ghstack-poisoned]

zou3519 committed:
Update on "Install networkx when testing functorch"


networkx is a required testing dependency. We're working on getting rid
of the dependency though (because ultimately we don't want PyTorch to
take a dependency on it).

Test Plan:
- wait for the functorch test shard in pytorch CI

[ghstack-poisoned]

malfet commented on .circleci/docker/requirements-ci.txt:
```diff
@@ -86,11 +86,11 @@ mypy==0.960
 #the structure, dynamics, and functions of complex networks
-#Pinned versions: 2.0
-#test that import:
+#Pinned versions: 2.6.3 (latest version that works with Python 3.7+)
```
requirements allow you to specify multiple version depending on Python version, for example see 
https://github.com/pytorch/pytorch/blob/2f0172bfc4c151cbc08d186b67f0969d7efdf953/.circleci/docker/requirements-ci.txt#L101-L102

zou3519 replied:
I'd prefer to keep it at 2.6.3 just in case there are important differences between the networkx versions
